backend/src/v2/cmd/compiler/main.go

```diff
@@ -109,8 +109,12 @@ func loadSpec(path string) (*pipelinespec.PipelineJob, error) {
 		return nil, err
 	}
 	spec := &pipelinespec.PipelineSpec{}
-	if err := protojson.Unmarshal(bytes, spec); err != nil {
-		return nil, fmt.Errorf("Failed to parse pipeline spec, error: %s, spec: %v", err, string(bytes))
+	specJson, err := yaml.YAMLToJSON(bytes)
```

Comment:
No AI for now, but once we add comments to the yaml, this might fail. We probably need some preprocess here.

Reply:
Not sure about this, but if the comments are only at the beginning of the yaml, I recall the `yaml.YAMLToJSON()` function just omits it?